test/parallel/test-net-writable.js

```diff
@@ -0,0 +1,16 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+
+const net = require('net');
+
+net.createServer(common.mustCall(function(s) {
+  this.close();
+  s.end();
+})).listen(0, 'localhost', common.mustCall(function() {
+  const socket = net.connect(this.address().port, 'localhost');
+  socket.on('end', common.mustCall(() => {
+    assert.strictEqual(socket.writable, true);
```

Comment:
Shouldn't this be `assert.strictEqual(socket.writable, false);`? I think this is the regression described in https://github.com/nodejs/node/issues/36029.

Reply:
No, half open close enforcer runs in next tick.

Reply:
Ah, yes you are right.